src/main/groovy/net/neoforged/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -99,4 +113,46 @@ abstract class GradleUtilsExtension {
     Action<? extends MavenArtifactRepository> getMaven() {
         return GradleUtils.getForgeMaven()
     }
+
+    void setupCentralPublishing() {
+        if (project.rootProject.name != project.name) {
+            throw new UnsupportedOperationException('The nexus publishing plugin can only be applied on the root project!')
+        }
+
+        project.plugins.apply(NexusPublishPlugin)
+        project.extensions.configure(NexusPublishExtension) { ext ->
+            ext.repositories { container ->
+                container.sonatype {
+                    it.username.set(System.getenv('SONATYPE_USER') ?: '')
+                    it.password.set(System.getenv('SONATYPE_PASSWORD') ?: '')
+                    it.nexusUrl.set(URI.create('https://s01.oss.sonatype.org/service/local/'))
+                }
+            }
+        }
+    }
+
+    void sign(Publication publication) {
+        project.afterEvaluate {
+            if (shouldSign.get()) {
+                project.extensions.configure(SigningExtension) { ext ->
+                    ext.sign(publication)
+                }
+            }
+        }
+    }
+
+    void setupSigning(Project project = this.project) {
+        project.plugins.apply(SigningPlugin)
```

Comment:
I'm mostly of the opinion that applying the plugin should be at the discretion of the buildscript author, and that GU should configure if present. I'll think on it, but what reasons are there for GU to apply the plugin itself here?